### PR TITLE
[RESTEASY-1565]  (master)

### PR DESCRIPTION
**Pull request by ronsigal**
Handle produced text-like media types without a charset.

### CODE REVIEW

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/resource/basic/DefaultCharsetTest.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+    protected enum ADD_CHARSET {TRUE, FALSE, DEFAULT};
+
+    static ResteasyClient client;
+
+    @Deployment(name = "true") // "resteasy.add.charset" set to true
+    public static Archive<?> deploy_charset() {
+        WebArchive war = TestUtil.prepareArchive(DefaultCharsetTest.class.getSimpleName() + "_true");
+        war.addAsWebInfResource(DefaultCharsetTest.class.getPackage(), "DefaultCharsetTestWeb_true.xml", "web.xml");
+        System.out.println(war.toString(true));
```
Comment (asoldano) [inline testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/resource/basic/DefaultCharsetTest.java]: Not really a major problem, but can we get rid of System.out? :-)

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/resource/basic/DefaultCharsetTest.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        return TestUtil.finishContainerPrepare(war, null, DefaultCharsetResource.class);
+    }
+    
+    @Deployment(name = "false") // "resteasy.add.charset" set to false
+    public static Archive<?> deploy_nocharset() {
+        WebArchive war = TestUtil.prepareArchive(DefaultCharsetTest.class.getSimpleName() + "_false");
+        war.addAsWebInfResource(DefaultCharsetTest.class.getPackage(), "DefaultCharsetTestWeb_false.xml", "web.xml");
+        System.out.println(war.toString(true));
```
Comment (asoldano) [inline testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/resource/basic/DefaultCharsetTest.java]: Same as above

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/resource/basic/DefaultCharsetTest.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        return TestUtil.finishContainerPrepare(war, null, DefaultCharsetResource.class);
+    }
+    
+    @Deployment(name = "default") // "resteasy.add.charset" not set
+    public static Archive<?> deploy_default() {
+        WebArchive war = TestUtil.prepareArchive(DefaultCharsetTest.class.getSimpleName() + "_default");
+        war.addAsWebInfResource(DefaultCharsetTest.class.getPackage(), "DefaultCharsetTestWeb_default.xml", "web.xml");
+        System.out.println(war.toString(true));
```
Comment (asoldano) [inline testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/resource/basic/DefaultCharsetTest.java]: Same as above

**Comment by asoldano** [comment]
@ronsigal , just a FYI, you can use github labels (on the right on this page) to help figuring out which branch a PR is for.

**Comment by ronsigal** [comment]
@asoldano 

re " if this is the latest solution agreed on the jira"

It's as close as we'll come to agreeing. ;-)

re: "you can use github labels"

Ah, I've seen indications that you were adding labels somewhere, but I didn't know what you were doing until now. Great.

**Comment by asoldano** [comment]
Merged, thanks

**Pull request merged**